File: modules/agent/src/models/SalaryRange.py
"""SalaryRange model for job salary brackets."""

from sqlalchemy import Column, Text, DateTime, BigInteger, Integer, ForeignKey, func
from models import Base


class SalaryRange(Base):
    """SalaryRange SQLAlchemy model."""

    __tablename__ = "Salary_Range"

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    label = Column(Text, nullable=False, unique=True)
    min_value = Column(Integer, nullable=True)
    max_value = Column(Integer, nullable=True)
    display_order = Column(Integer, nullable=False, default=0)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)
    created_by = Column(BigInteger, ForeignKey("User.id"), nullable=False)
    updated_by = Column(BigInteger, ForeignKey("User.id"), nullable=False)

File: modules/agent-go/internal/models/SalaryRange.go
package models

import "time"

type SalaryRange struct {
	ID           int64     `gorm:"primaryKey;autoIncrement" json:"id"`
	Label        string    `gorm:"uniqueIndex;not null" json:"label"`
	MinValue     *int      `json:"min_value"`
	MaxValue     *int      `json:"max_value"`
	DisplayOrder int       `gorm:"not null;default:0" json:"display_order"`
	CreatedAt    time.Time `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt    time.Time `gorm:"autoUpdateTime" json:"updated_at"`
}

func (SalaryRange) TableName() string {
	return "Salary_Range"
}
